package/gluon-xlat464-plat/files/xlat464plat.sh

```diff
@@ -0,0 +1,172 @@
+
+  if [ ${debug} -ne 0 ]; then
+    plat_cfg="${plat_cfg} debug 1"
+  fi
+
+  plat_cfg="${plat_cfg} local.style RFC6052 local.v6 ${local_v6} local.v4 ${local_v4}"
+  plat_cfg="${plat_cfg} remote.style NONE remote.v4 192.0.0.8/32 remote.v6 ${nat66addr}/128"
+
+  echo "add ${config}" > ${CTR}
+  echo "${plat_cfg}" > ${CTR}
+
+
+  proto_init_update "${config}" 1
+
+  proto_add_ipv6_address "${local_v6%%:/*}c000:1" "128"
+  proto_add_ipv6_route "${local_v6%%/*}" "${local_v6##*/}"
+  proto_add_ipv4_route "192.0.0.8" "32"
+
+	proto_add_data
+	[ "$zone" != "-" ] && json_add_string zone "$zone"
+
+	json_add_array firewall
+		# this fw3 IPv6 SNAT rule is broken
+		json_add_object ""
+			json_add_string type nat
+			json_add_string target SNAT
+			json_add_string family ipv6
+			json_add_string snat_ip "${nat66addr}"
+		json_close_object
+
+	  if [ "$zone" != "-" ]; then
+	  	json_add_object ""
+	  		json_add_string type rule
+	  		json_add_string family inet
+	  		json_add_string proto all
+	  		json_add_string direction in
+			json_add_string dest "$zone"
+			json_add_string src "$zone"
+	  		json_add_string src_ip 192.0.0.8
+	  		json_add_string target ACCEPT
+	  	json_close_object
+	  fi
+
+	  if [ "$zone6" != "-" ]; then
+		# deny access to 192.168.0.0/16
+	  	json_add_object ""
+	  		json_add_string type rule
+	  		json_add_string family ipv6
+	  		json_add_string proto all
+	  		json_add_string direction out
+			json_add_string dest "$zone"
+			json_add_string src "${zone6}"
+	  		json_add_string dest_ip "${local_v6%%:/*}c0a8::/112"
+	  		json_add_string target DROP
+	  	json_close_object
+		# deny access to 172.16.0.0/12
+		json_add_object ""
+	  		json_add_string type rule
+	  		json_add_string family ipv6
+	  		json_add_string proto all
+	  		json_add_string direction out
+			json_add_string dest "$zone"
+			json_add_string src "${zone6}"
+	  		json_add_string dest_ip "${local_v6%%:/*}ac10::/108"
+	  		json_add_string target DROP
+	  	json_close_object
+		# deny access to 10.0.0.0/8
+	  	json_add_object ""
+	  		json_add_string type rule
+	  		json_add_string family ipv6
+	  		json_add_string proto all
+	  		json_add_string direction out
+			json_add_string dest "$zone"
+			json_add_string src "${zone6}"
+	  		json_add_string dest_ip "${local_v6%%:/*}a00::/104"
+	  		json_add_string target DROP
+	  	json_close_object
+		# allow access to everything else
+	  	json_add_object ""
+	  		json_add_string type rule
+	  		json_add_string family ipv6
+	  		json_add_string proto all
+	  		json_add_string direction out
+			json_add_string dest "$zone"
+			json_add_string src "${zone6}"
+	  		json_add_string dest_ip "${local_v6}"
+	  		json_add_string target ACCEPT
+	  	json_close_object
+	  fi
+	json_close_array
+
+	proto_close_data
+
+  proto_send_update "${config}"
+}
+
+
+proto_xlat464plat_teardown() {
+  local config="$1"
+  echo "del ${config}" > ${CTR}
+}
+
+
+[ -n "$INCLUDE_ONLY" ] || {
+  add_protocol xlat464plat
+}
+
```

Comment:
empty line at EoF - and please check the whole file for double/triple empty lines